# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/nsDocShell.cpp
# Commit: d2fd0f955e24
# Full Hash: d2fd0f955e2462852c0a37ca73267b20ac0be9df
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-07-13 09:51:22
# Regressor Bug: 1646899
# File Overlap Count: 1
# Description:
#   Bug 1646899 - P3: Modify CanUseDocumentChannel to take necessary vars. r=mattwoodrow
#   
#   Allows checking if DocumentChannel can be used with out creating an
#   nsDocShellLoadState object.
#   
# ==============================================================================

diff -r 7a7194146fe1 -r d2fd0f955e24 docshell/base/nsDocShell.cpp
--- a/docshell/base/nsDocShell.cpp	Mon Jul 13 00:48:52 2020 +0000
+++ b/docshell/base/nsDocShell.cpp	Mon Jul 13 00:48:54 2020 +0000
@@ -8691,7 +8691,8 @@
   // Similar check will be performed by the ParentProcessDocumentChannel if in
   // use.
   if (XRE_IsE10sParentProcess() &&
-      !DocumentChannel::CanUseDocumentChannel(aLoadState) &&
+      !DocumentChannel::CanUseDocumentChannel(aLoadState->URI(),
+                                              aLoadState->LoadFlags()) &&
       !CanLoadInParentProcess(aLoadState->URI())) {
     return NS_ERROR_FAILURE;
   }
@@ -9571,7 +9572,8 @@
       mBrowsingContext, Some(uriModified), Some(isXFOError));
 
   nsCOMPtr<nsIChannel> channel;
-  if (DocumentChannel::CanUseDocumentChannel(aLoadState)) {
+  if (DocumentChannel::CanUseDocumentChannel(aLoadState->URI(),
+                                             aLoadState->LoadFlags())) {
     channel = DocumentChannel::CreateDocumentChannel(aLoadState, loadInfo,
                                                      loadFlags, this, cacheKey,
                                                      uriModified, isXFOError);